python/tank/platform/engine.py

```diff
@@ -175,6 +175,13 @@ def __init__(self, tk, context, engine_instance_name, env):
         qt_abstraction.QtCore = qt.QtCore
         qt_abstraction.QtGui = qt.QtGui
         qt_abstraction.QtWidgets = qt.QtWidgets
+
+        # Setup the engine base class event signal. This isn't emitted by
+        # the engine base class itself, but is provided for derived engine
+        # classes and can be used to broadcast event notifications to other
+        # bundles, like apps.
+        from .qt import event_emitter
+        self._emitter = event_emitter.EventEmitter()
```

Comment:
i guess if we switched to a simpler callback system, we would write our own simple emitter which would just round-robin over all items in `self.apps` and `self.frameworks`